vm/src/stdlib/itertools.rs

```diff
@@ -235,7 +240,12 @@ mod decl {
         fn next(zelf: &Py<Self>, vm: &VirtualMachine) -> PyResult<PyIterReturn> {
             let mut cur = zelf.cur.write();
             let result = cur.clone();
-            *cur += &zelf.step;
+            if let Some(step) = &zelf.step {
+                *cur = vm._iadd(&*cur, step.as_object())?;
+            } else {
+                let one = vm.new_pyobj(1);
+                *cur = vm._add(&*cur, &one)?;
+            }
```

Comment:
```suggestion
            *cur = vm._add(&*cur, &zelf.step)?;
```
If you have interest, this line could be replaced with `PyNumber::add` which is not yet implemented in rustpython

Reply:
that will not be easy before #3819